site/components/map/Map.tsx

```diff
@@ -261,14 +361,137 @@ const Map = ({
         setSelectAll(!selectAll);
     };
 
+    useEffect(() => {
+        setSelectAll(true);
+    }, [searchedRoutes]);
```

Comment:
Why are we setting selectAll to be true when searchedRoutes changes?

Reply:
Let's say I have clicked the select all button before when selecting a service - if a service changes its searchedRoutes will change. To prevent the button from staying on 'unselect all' I change reset it back to 'select all'.

Reply:
Ah - I didnt realise selectAll (boolean) was being used for the text of the button. Can we just check that all the searchedRoutes are selected to decide what text to have the button as?

Reply:
Yeah just reread the file - when the component is first rendered, selectAll is true - id expect selectAll to be true if the user has clicked it or has manually selected all the options.

Reply:
So when the user clicks the button, they have chosen to selectAll, so selectAll would be true.

Reply:
it would be hard to check if all the routes are selected due to the 100 stops limitation

Reply:
Yeah fair enough - but I do think the value should be reversed. If the user comes into the page, and they havent selected all - then selectAll should be false. Having it be true, and checking for !selectAll is a little confusing.

Reply:
done